Realm/Realm/Native/SessionNotificationToken.cs

```diff
@@ -0,0 +1,25 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+namespace Realms.Sync.Native
+{
+    internal struct SessionNotificationToken
```

Comment:
```suggestion

using System.Runtime.InteropServices;

namespace Realms.Sync.Native
{
    [StructLayout(LayoutKind.Sequential)]
    internal struct SessionNotificationToken
```
We decorate all structs that we pass to and from native with this attribute in order to enforce that the runtime layout of the struct matches the layout of the corresponding struct in native. The C# compiler generates sequential structs by default, but our convention is to be explicit about marshaling declarations.